tests/test_convert_text_to_mds.py

```diff
@@ -0,0 +1,198 @@
+        input_folder=f's3://fake-test-input-path',
+        concat_tokens=2048,
+        eos_text='',
+        bos_text='',
+        no_wrap=False,
+        max_workers=1,
+        compression='zstd',
+        processes=processes,
+        args_str='Namespace()',
+        reprocess=False,
+    )
+
+
+# Mock starmap with no multiprocessing
+def _mock_starmap(func: Callable, args: Iterable):
+    for arg in args:
+        func(*arg)
+
+
+# Build a dataloader with no threading so mock call counts are correct
+def _mock_build_dataloader(dataset: Dataset,
+                           batch_size: int,
+                           num_workers: Optional[int] = None) -> DataLoader:
+    return DataLoader(
+        dataset=dataset,
+        sampler=None,
+        batch_size=batch_size,
+        num_workers=0,
+    )
+
+
+def _assert_files_exist(prefix: str, files: List[str]):
+    for file in files:
+        assert os.path.exists(os.path.join(prefix, file))
+
+
+@pytest.mark.parametrize('processes', [1, 2, 3])
+@patch('scripts.data_prep.convert_text_to_mds.build_dataloader',
+       new=Mock(wraps=_mock_build_dataloader))
+@patch.object(Pool, 'starmap', new=Mock(wraps=_mock_starmap))
+@patch(
+    'scripts.data_prep.convert_text_to_mds.maybe_create_object_store_from_uri')
+@patch('scripts.data_prep.convert_text_to_mds.parse_uri')
+@patch('scripts.data_prep.convert_text_to_mds.download_and_convert',
+       wraps=download_and_convert)
+@patch('scripts.data_prep.convert_text_to_mds.merge_shard_groups',
+       wraps=merge_shard_groups)
+def test_single_and_multi_process(merge_shard_groups: Mock,
+                                  download_and_convert: Mock, parse_uri: Mock,
+                                  maybe_create_object_store_from_uri: Mock,
+                                  tmp_path: pathlib.Path, processes: int):
+    remote_folder = os.path.join(tmp_path, 'remote')
+    text_content = 'HELLO WORLD'
+    tokenizer_name = 'mosaicml/mpt-7b'
+    n_text_files = processes * 3
+
+    mock_object_store = Mock(
+        wraps=MockObjectStore(remote_folder, n_text_files, text_content))
+    maybe_create_object_store_from_uri.return_value = mock_object_store
+    parse_uri.return_value = ('s3', 'fake-test-bucket', str(remote_folder))
+
+    _call_convert_text_to_mds(processes=processes,
+                              tokenizer_name=tokenizer_name)
+
+    # Check call counts
+    assert download_and_convert.call_count == processes  # called once per process
+    assert mock_object_store.download_object.call_count == n_text_files + 1  # text files + done file
+    assert mock_object_store.upload_object.call_count == processes + 2  # shard per process + done file + index.json
+
+    if processes > 1:
+        merge_shard_groups.assert_called_once()
+
+    total_object_names = 0
+    for call_args in download_and_convert.call_args_list:
+        object_names = call_args[0][0]
+        assert (len(object_names) == n_text_files / processes
```

Comment:
can we test when this is ragged too? like object names doesn't evenly divide into processes groups